drivers/boston_project_scorecard/app/controllers/boston_project_scorecard/warehouse_reports/scorecards_controller.rb

```diff
@@ -0,0 +1,280 @@
+###
+# Copyright 2016 - 2022 Green River Data Analysis, LLC
+#
+# License detail: https://github.com/greenriver/hmis-warehouse/blob/production/LICENSE.md
+###
+
+module BostonProjectScorecard::WarehouseReports
+  class ScorecardsController < ApplicationController
+    include WarehouseReportAuthorization
+    include ArelHelper
+    # TODO: what are the access rules?
+    before_action :set_filter, only: [:index, :create]
+    before_action :set_projects_and_groups, :set_current_reports, only: [:index]
+    before_action :set_report, only: [:show, :edit, :rewind, :complete, :update]
+    before_action :set_history_filter, only: [:history]
+
+    def index
+      start_date = Date.current.prev_month.beginning_of_month
+      end_date = Date.current.prev_month.end_of_month
+      @range = ::Filters::FilterBase.new(start: start_date, end: end_date, user_id: current_user.id)
+    end
+
+    def history
+      scope = filtered_reports_scope.
+        order(id: :desc).
+        preload(project: [:organization, :data_source])
+      @pagy, @reports = pagy(scope, items: 50)
+    end
+
+    private def available_users
+      creator_user_ids = reports_scope.distinct.pluck(:user_id)
+      User.active.where(id: creator_user_ids)
+    end
+    helper_method :available_users
+
+    def show
+      report_id = { report_id: @report.id }
+      @pdf_export = BostonProjectScorecard::DocumentExports::ScorecardExport.new(query_string: report_id.to_query, user: current_user)
+    end
+
+    def create
+      @range = ::Filters::FilterBase.new(user_id: current_user.id)
+      @range.set_from_params(scorecard_filter_params)
+      errors = @range.errors.messages.map { |k, v| "#{k}: #{v.join(', ')}".humanize }
+
+      @project_ids = params[:project]&.keys&.map(&:to_i) || []
+      @project_group_ids = params[:project_group]&.keys&.map(&:to_i) || []
```

Comment:
is it worth standardizing the parameter names on the form?  It seems like a copy/paste legacy that we have :project and :project_group instead of :project_ids and :project_group_ids.